src/ServerScriptService/InstanceComponents/Misc/Sprint.lua

```diff
@@ -0,0 +1,50 @@
+local Players = game:GetService("Players")
+local ReplicatedStorage = game:GetService("ReplicatedStorage")
+local Trove = require(ReplicatedStorage.Packages.Trove)
+local Comm = require(ReplicatedStorage.Packages.Comm)
+local WalkSpeedConfig = require(ReplicatedStorage.Common.Config.WalkSpeed)
+
+local Sprint = {}
+Sprint.__index = Sprint
+
+function Sprint.new(character: Model)
+	local serverComm = Comm.ServerComm.new(character, "Sprint")
+	local self = setmetatable({}, Sprint)
+
+	self._comm = serverComm
+	self._trove = Trove.new()
+	self._trove:Add(self._comm)
+	self._owner = Players:GetPlayerFromCharacter(character.Parent)
+	self.Instance = character
+
+	self._trove:Connect(character.AncestryChanged, function()
+		self._owner = Players:GetPlayerFromCharacter(character.Parent)
+	end)
+
+	self._startSprinting = serverComm:WrapMethod(self, "StartSprinting")
+	self._stopSprinting = serverComm:WrapMethod(self, "StopSprinting")
```

Comment:
WrapMethod is really cool! Are we actually using what it returns, though? It looks like we're just using the serverObject anyways.

Reply:
Nope, these aren't used. This is also why the `:IsOwner()` thing is the way it is, as the methods are designed to be public for the jetpack and sprint code but also able to be called by the client.